api/envoy/extensions/certificate_providers/local_certificate/v3/local_certificate.proto

```diff
@@ -30,4 +32,7 @@ message LocalCertificate {
 
     // Indicates the time at which the certificate expires.
   google.protobuf.Timestamp expiration_time = 5;
+
+  // The maximum number of hosts that the cache will hold. If not specified defaults to 1024.
+  google.protobuf.UInt32Value max_hosts = 6 [(validate.rules).uint32 = {gt: 0}];
```

Comment:
The cache is for certs, "max_hosts" sounds we need to cache host names, we can tweak the name a little bit, like "cert_cache_size"

Reply:
OK, sounds good